tensorflow/compiler/mlir/hlo/lib/Dialect/mhlo/IR/hlo_ops.cc

```diff
@@ -1547,6 +1547,51 @@ static LogicalResult Verify(DynamicSliceOp op) {
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+// RealDynamicSliceOp
+//===----------------------------------------------------------------------===//
+// Verifies that operand rank matches start_indices/limit_indices/strides size
+static LogicalResult Verify(RealDynamicSliceOp op) {
+  auto input_type = op.operand().getType().dyn_cast<RankedTensorType>();
+  if (!input_type) {
+    return op.emitOpError() << "operand is not RankedTensorType";
+  }
+  int input_rank = input_type.getRank();
+
+  auto start_type = op.start_indices().getType().dyn_cast<RankedTensorType>();
+  auto limit_type = op.limit_indices().getType().dyn_cast<RankedTensorType>();
+  auto strides_type = op.strides().getType().dyn_cast<RankedTensorType>();
+
+  if (!start_type || start_type.getRank() != 1) {
+    return op.emitOpError() << "start_indices's rank must be 1.";
+  }
```

Comment:
We should be able to eliminate this check with a more suitable type in ODS (see my comment there)

Reply:
Done. We change these arguments' type to HLO_DimensionTensor and eliminate these checks.